azure_functions_worker/utils/dependency.py

```diff
@@ -0,0 +1,344 @@
+        on existing customers.
+
+        Only intended to be used in Linux Consumption scenario.
+        """
+        # Reload package namespaces for customer's libraries
+        packages_to_reload = ['azure', 'google']
+        for p in packages_to_reload:
+            try:
+                logger.info(f'Reloading {p} module')
+                importlib.reload(sys.modules[p])
+            except Exception as ex:
+                logger.info('Unable to reload {}: \n{}'.format(p, ex))
+            logger.info(f'Reloaded {p} module')
+
+        # Reload azure.functions to give user package precedence
+        logger.info('Reloading azure.functions module at %s',
+                    inspect.getfile(sys.modules['azure.functions']))
+        try:
+            importlib.reload(sys.modules['azure.functions'])
+            logger.info('Reloaded azure.functions module now at %s',
+                        inspect.getfile(sys.modules['azure.functions']))
+        except Exception as ex:
+            logger.info('Unable to reload azure.functions. '
+                        'Using default. Exception:\n{}'.format(ex))
+
+    @classmethod
+    def reload_all_namespaces_from_customer_deps(cls, cx_working_dir: str):
+        """This is a new implementation of reloading azure and google
+        namespace from customer's .python_packages folder. Only intended to be
+        used in Linux Consumption scenario.
+
+        Parameters
+        ----------
+        cx_working_dir: str
+            The path which contains customer's project file (e.g. wwwroot).
+        """
+        # Specialized working directory needs to be added
+        working_directory: str = os.path.abspath(cx_working_dir)
+
+        # Switch to customer deps and clear out all module cache in worker deps
+        cls._remove_from_sys_path(cls.worker_deps_path)
+        cls._add_to_sys_path(cls.cx_deps_path, True)
+        cls._add_to_sys_path(working_directory, False)
+        logger.info('Reloaded azure google namespace from customer deps')
+
+    @classmethod
+    def _add_to_sys_path(cls, path: str, add_to_first: bool):
+        """This will ensure no duplicated path are added into sys.path and
+        clear importer cache. No action if path already exists in sys.path.
+
+        Parameters
+        ----------
+        path: str
+            The path needs to be added into sys.path.
+            If the path is an empty string, no action will be taken.
+        add_to_first: bool
+            Should the path added to the first entry (highest priority)
+        """
+        if path and path not in sys.path:
+            if add_to_first:
+                sys.path.insert(0, path)
+            else:
+                sys.path.append(path)
+
+            # Only clear path importer and sys.modules cache if path is not
+            # defined in sys.path
+            cls._clear_path_importer_cache_and_modules(path)
+
+    @classmethod
+    def _remove_from_sys_path(cls, path: str):
```

Comment:
It would have made it a little simpler if the path was an array of strings rather than a string. You don't need to change it, but something to think about later - not needed now.